ReactiveCocoa/Swift/Signal.swift

```diff
@@ -177,6 +177,24 @@ public func filter<T, E>(predicate: T -> Bool)(signal: Signal<T, E>) -> Signal<T
 	}
 }
 
+/// Unwraps non-`nil` values from `signal` and forwards them on the returned
+/// signal, `nil` values are dropped.
+public func ignoreNil<T, E>(signal: Signal<T?, E>) -> Signal<T, E> {
+	return Signal { observer in
+		signal.observe(next: { value in
+			if let val = value {
+				sendNext(observer, val)
+			}
+		}, error: { error in
+			sendError(observer, error)
+		}, completed: {
+			sendCompleted(observer)
+		}, interrupted: {
+			sendInterrupted(observer)
+		})
+	}
```

Comment:
Perhaps it might be easier to understand by just composing existing operators like:

```swift
return signal 
           |> filter { $0 != nil }
           |> map { $0! }
```